src/platform/switch/main.c

```diff
@@ -0,0 +1,486 @@
+/* Copyright (c) 2013-2015 Jeffrey Pfau
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <mgba/core/blip_buf.h>
+#include <mgba/core/core.h>
+#include <mgba/core/serialize.h>
+#ifdef M_CORE_GBA
+#include <mgba/internal/gba/gba.h>
+#include <mgba/internal/gba/input.h>
+#include <mgba/internal/gba/video.h>
+#endif
+#ifdef M_CORE_GB
+#include <mgba/internal/gb/gb.h>
+#endif
+#include "feature/gui/gui-runner.h"
+#include <mgba-util/gui.h>
+#include <mgba-util/gui/file-select.h>
+#include <mgba-util/gui/font.h>
+#include <mgba-util/gui/menu.h>
+#include <mgba-util/memory.h>
+
+#include "nx-gfx.h"
+#include <mgba-util/threading.h>
+
+#include <switch.h>
+
+mLOG_DECLARE_CATEGORY(GUI_SWITCH);
+mLOG_DEFINE_CATEGORY(GUI_SWITCH, "Switch", "gui.switch");
+
+static enum DarkenMode { DM_NATIVE, DM_MULT, DM_MULT_SCALE, DM_MULT_SCALE_BIAS, DM_MAX } darkenMode = DM_NATIVE;
+
+#define _SWITCH_INPUT 0x3344535C
+
+#define AUDIO_SAMPLES 384
+#define AUDIO_SAMPLE_BUFFER (AUDIO_SAMPLES * 16)
+#define DSP_BUFFERS 4
+
+static enum { NO_SOUND, DSP_SUPPORTED } hasSound;
+
+enum {
+	KEY_GENERIC_LEFT = BIT(26), // collides with KEY_TOUCH
+	KEY_GENERIC_UP = BIT(27),
+	KEY_GENERIC_RIGHT = BIT(28),
+	KEY_GENERIC_DOWN = BIT(29)
+};
+
+static enum {
+	SM_ORIGINAL,
+	SM_NEAREST2X,
+	SM_NEAREST3X,
+	SM_NEAREST4X,
+	SM_NEARESTMAX,
+	SM_COUNT
+} screenMode = SM_NEARESTMAX;
+
+// TODO: Move into context
+static void* outputBuffer;
+#define OUTPUT_IMAGE_W 256
+#define OUTPUT_IMAGE_H 224
+static nxImage outputImage;
+
+static struct mAVStream stream;
+static int16_t* audioLeft = 0;
+static int16_t* audioRight = 0;
+static size_t audioPos = 0;
+// static ndspWaveBuf dspBuffer[DSP_BUFFERS];
+static int bufferId = 0;
+
+static unsigned* mGBAscreenWidth = NULL;
```

Comment:
Better names please. `screenWidth` is probably fine since it's static, but why are they pointers?

Reply:
they are pointers because they are used change the GUI runner params width and height parameters, so that the GUI adapts to the changing resolution between portable and TV-mode. I haven't found a better way to do this.
See here:
https://github.com/mgba-emu/mgba/pull/1084/files/9588b7c1432a9ae22a85ff871876e2589727ef9f#diff-4a2fe7df95d98ca31660e48730e04fbbR459

Reply:
Check out what the Wii version does: https://github.com/mgba-emu/mgba/blob/master/src/platform/wii/main.c#L222-L223